pkg/server/server.go

```diff
@@ -649,9 +677,19 @@ loop:
 				Callbacks: leaderelection.LeaderCallbacks{
 					OnStartedLeading: func(leaderElectionCtx context.Context) {
 						electionLogger.Info("started leading")
+						if len(s.controllers) == 0 {
+							s.controllerLock.Lock()
+							defer s.controllerLock.Unlock()
```

Comment:
IIUC the installation of controllers happen only in one thread. Do we need to acquire a lock for that?

Reply:
This was added , so there would be no race happens when the uninstallController and installController